flumes/discoverer.py

```diff
@@ -364,6 +365,15 @@ def on_file_found(self, enum, res, user_data):
     def on_directory_content(self, path, res, user_data):
         enum = path.enumerate_children_finish(res)
         enum.next_files_async(1, 0, None, self.on_file_found, None)
+    
+    def on_signal_handler(self, signum, frame):
```

Comment:
The signal handler is done through python signals package, can't be done through GLib's signal handler? Given that the whole code is already using that?